src/editor/Gui.cpp

```diff
@@ -3740,6 +4258,36 @@ void Gui::drawTextureTool() {
 						break;
 					}
 				}
+
+				if (!validTexture)
+				{
+					for (auto & s : mapRenderer->wads)
+					{
+						if (s->hasTexture(textureName))
+						{
+							WADTEX* wadTex = s->readTexture(textureName);
+							int lastMipSize = (wadTex->nWidth / 8) * (wadTex->nHeight / 8);
+
+							COLOR3* palette = (COLOR3*)(wadTex->data + wadTex->nOffsets[3] + lastMipSize + 2 - 40);
+							BYTE* src = wadTex->data;
+
+							COLOR3* imageData = new COLOR3[wadTex->nWidth * wadTex->nHeight];
+
+							int sz = wadTex->nWidth * wadTex->nHeight;
+
+							for (int k = 0; k < sz; k++) {
+								imageData[k] = palette[src[k]];
+							}
+							map->add_texture(textureName, (BYTE *)imageData, wadTex->nWidth, wadTex->nHeight);
+							delete[] imageData;
+							delete wadTex;
+							mapRenderer->reloadTextures();
+							textureChanged = true;
+							ImGui::End();
+							return;
+						}
+					}
+				}

```

Comment:
I don't think selecting a missing texture should automatically create one. Why do this?

Reply:
@wootguy i think this code create texture in map if not found.